kubemarine/procedures/check_paas.py

```diff
@@ -308,10 +313,54 @@ def thirdparties_hashes(cluster):
                 # if thirdparty is missing somewhere, do not check anything further for it
                 continue
 
-            if 'sha1' not in config:
-                # silently skip if SHA not defined
-                continue
+            is_curl = config['source'][:4] == 'http' and '://' in config['source'][4:8]
+            expected_sha = None
+
+            # Get sha from source, if it can be downloaded
+            if is_curl:
+                cluster.log.verbose(f"Thirdparty {path} doesn't have default sha, download it...")
+                #Create tmp dir for loading thirdparty without default sha
+                random_dir = "/tmp/%s" % uuid.uuid4().hex
+                final_commands = "rm -r -f %s" % random_dir
+                random_path = "%s%s" % (random_dir, path)
+                cluster.log.verbose('Temporary path: %s' % random_path)
+                remote_commands = "mkdir -p %s" % ('/'.join(random_path.split('/')[:-1]))
+                #Load thirdparty to temporary dir
+                remote_commands += "&& sudo curl -f -g -s --show-error -L %s -o %s" % (config['source'], random_path)
+                results = first_control_plane.sudo(remote_commands, hide=True, warn=True)
+                for host, result in results.items():
+                    if result.failed:
+                        broken.append(f"Can`t download thirdparty {path} on {host.host} for getting sha: {result.stderr}")
+                        cluster.log.verbose(f"Can`t download thirdparty {path} on {host.host} for getting sha: {result.stderr}")
+                if not results.is_any_failed():
+                    #Get temporary thirdparty sha
+                    cluster.log.verbose(f"Get temporary thirdparty sha for {path}...")
+                    results = first_control_plane.sudo(f'openssl sha1 {random_path} | sed "s/^.* //"', warn=True)
+                    for host, result in results.items():
+                        if result.failed:
+                            broken.append(f'failed to get sha for temporary file {random_path} on {host.host}: {result.stderr}')
+                            cluster.log.verbose(f'failed to get sha for temporary file {random_path} on {host.host}: {result.stderr}')
+                        else:
+                            expected_sha = result.stdout.strip()
+                            cluster.log.verbose(f"Expected sha was got for {path}")
+                # Remove temporary dir in any case
+                cluster.log.verbose(f"Remove temporary dir {random_dir}...")
+                first_control_plane.sudo(final_commands, hide=True, warn=True)
 
+            recommended_sha = thirdparties.get_thirdparty_recommended_sha(path, cluster)
```

Comment:
Now we have two independent algorithms to calculate recommended sha. In defaults using jinja templates and in python code. So why not keep the single algorithm then? Just remove sha sections from defaults.yaml and enrich them in thirdparties.enrich_inventory_apply_defaults

Reply:
Default sha were calculated from function. sha fields from defaults.yaml were deleted